### PR TITLE
Support Struct Data Query In SQL/PPL

### PR DESCRIPTION
**Pull request by penghuo**
*Issue #, if available:* #795

*Description of changes:*
1. Support Struct Data Query In SQL/PPL. https://github.com/penghuo/sql/blob/object-query-pr/docs/experiment/ppl/general/datatypes.rst#query-struct-data-types
2. Add the Paths in ReferenceExpression.

## To Reviewer
There are 2 ITs in legacy engine have been disabled.
1. ObjectFieldSelectIT.testSelectObjectFieldOfArrayValuesItself, the issue is tracking by #1017 
2. PrettyFormatResponseIT. the issue is tracking by #1019

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/ReferenceExpression.java [hunk 1]
```diff
@@ -49,4 +69,44 @@ public ExprType type() {
   public String toString() {
     return attr;
   }
+
+  /**
+   * Resolve the ExprValue from {@link ExprTupleValue} using paths.*
```
Comment (dai-chen) [inline core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/ReferenceExpression.java:74]: np: is `paths.*` intentional or a typo?

File: core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/ReferenceExpression.java [hunk 2]
```diff
@@ -49,4 +69,44 @@ public ExprType type() {
...
+   * 2. Resolve the path recursively through ExprValue.
+   *
+   * @param value {@link ExprTupleValue}.
+   * @return {@link ExprTupleValue}.
+   */
+  public ExprValue resolve(ExprTupleValue value) {
+    return resolve(value, paths);
+  }
+
+  private ExprValue resolve(ExprValue value, List<String> paths) {
```
Comment (dai-chen) [inline core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/ReferenceExpression.java:104]: Can this / do we need to handle `project.year` is another object field? ex. `project.year`.month

**Review by dai-chen — APPROVED** [review]
Just minor comments, thanks for the changes!

File: core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/ReferenceExpression.java (hunk 1 above)
Comment (penghuo) [reply core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/ReferenceExpression.java]: typo. removed.

File: core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/ReferenceExpression.java (hunk 2 above)
Comment (penghuo) [reply core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/ReferenceExpression.java:111]: Good point. add more cases in doc and UT.
We don't expect object filed name include "." which is not supported now.

**Review by chloe-zh — APPROVED** [review]

**Pull request merged**